helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -200,14 +196,15 @@ public void computeScore(Map<String, Integer> overallClusterRemainingCapMap) {
       // score = SUM(weight * (resource_capacity/cluster_capacity) where weight = 1/(1-total_util%)
       // it could be be simplified to "resource_capacity/cluster_remainingCapacity".
       for (Map.Entry<String, Integer> resourceCapacity : _replica.getCapacity().entrySet()) {
+        String capacityKey = resourceCapacity.getKey();
         if (resourceCapacity.getValue() == 0) {
           continue;
         }
-        score = (overallClusterRemainingCapMap.get(resourceCapacity.getKey()) == 0
+        score = (overallClusterRemainingCapMap.get(capacityKey) == 0
             || resourceCapacity.getValue() > (overallClusterRemainingCapMap
-            .get(resourceCapacity.getKey()))) ? Float.MAX_VALUE
+            .get(capacityKey))) ? Float.MAX_VALUE
             : score + (float) resourceCapacity.getValue() / (overallClusterRemainingCapMap
-                .get(resourceCapacity.getKey()));
+                .get(capacityKey));
         if (Float.compare(score, Float.MAX_VALUE) == 0) {
```

Comment:
@xyuanlu This is no longer needed since we are not doing Float.MAX_VALUE.

Reply:
Updated in new version.